external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -388,57 +406,50 @@ where
     v
 }
 
-/// Attempts to skip tokens until the end of the item in the comma-separated list (which started
-/// with an already consumed starting token) - looks for a matched ending token or after a comma.
-fn skip_comma_list_item(
+/// Attempts to skip tokens until the end of the item in the list with a given separator (which
+/// started with an already consumed starting token) - looks for a matched ending token or a token
+/// appearing after the separator. This helper function is also used to error advance sequences.
+fn list_error_advance(
     context: &mut Context,
     start_token: Tok,
     end_token: Tok,
+    sep_token: Tok,
+    for_list: bool,
     diag: Diagnostic,
-) -> bool {
+) {
     context.env.add_diag(diag);
-    let mut start_tokens: i32 = 0;
-    let mut stop_parsing = false;
+    let mut depth: i32 = 0; // When we find  another start token, we track how deep we are in them
     loop {
+        if for_list {
+            adjust_token(context.tokens, end_token);
+        }
```

Comment:
Can you explain what this flag is doing? When do we not want to adjust the tokens?

Reply:
Adjusting tokens only makes sense (and is only used) when parsing lists. It may not matter if used unconditionally also when parsing statements but it feels odd to use it in a case where it does not really belong (added a comment to that effect).